packages/itwin/ec3-widget/src/components/TemplateMenu.tsx

```diff
@@ -95,170 +56,144 @@ export const TemplateMenu = ({ template, onSaveSuccess, onClickCancel }: Templat
-      });
+  const saveConfiguration = async (): Promise<EC3Configuration | undefined> => {
+    try {
+      const token = await getAccessToken();
+      if (childTemplate.id) {
+        return await configurationsClient.updateConfiguration(token, childTemplate.id, childTemplate);
+      } else if (childTemplate.reportId) {
+        return await configurationsClient.createConfiguration(token, {
+          ...childTemplate,
+          reportId: childTemplate.reportId,
+        });
+      } else {
+        return undefined;
+      }
+    } catch (e) {
+      toaster.negative("Saving failed");
+      // eslint-disable-next-line no-console
+      console.error(e);
+      return undefined;
     }
-    return newReportOptions;
-  }, [availableReports]);
-
-  return (
-    <>
-      <div className="ec3w-template-details-container" data-testid="ec3-template-details">
-        <Fieldset legend="Template Details" className="ec3w-template-details">
-          <RequiredFieldsNotice />
-          <LabeledInput
-            id="templateName"
-            data-testid="ec3-template-name-input"
-            name="displayName"
-            label="EC3 Project Template Name"
-            value={childTemplate.displayName}
-            required
-            onChange={(event) => {
-              handleInputChange(event, childTemplate, setChildTemplate);
-            }}
-          />
-          <LabeledInput
-            id="templateDescription"
-            data-testid="ec3-template-description-input"
-            name="description"
-            label="Template description"
-            value={childTemplate.description}
-            onChange={(event) => {
-              handleInputChange(event, childTemplate, setChildTemplate);
-            }}
-          />
+  };
 
-          <div className="ec3w-report-select-container">
-            <div className="ec3w-report-select-combo-box">
-              <Label htmlFor="combo-input" required={true}>
-                Report
-              </Label>
-              {!template ? (
-                <ComboBox
-                  data-testid="ec3-enabled-selection"
-                  options={ReportOptions}
-                  value={childTemplate.reportId}
-                  onChange={async (value) => {
-                    if (childTemplate.labels.length > 0 && value !== childTemplate.reportId) {
-                      setPreviouslySelectedReport(childTemplate.reportId);
-                      setShowReportConfirmModal(true);
-                    }
-                    handleSelectChange(value, "reportId", childTemplate, setChildTemplate);
-                  }}
-                  inputProps={{
-                    id: "combo-input",
-                    placeholder: "Select report",
-                  }}
-                />
-              ) : (
-                <Select data-testid="ec3-disabled-selection" options={ReportOptions} value={childTemplate.reportId} disabled={true} />
-              )}
+  const renderSteps = () => {
```

Comment:
The whole `renderSteps` should become a separate component. It could be named `TemplateModificationStepRenderer`.

Reply:
done. Added 3 new components for the three steps